assertion/function/analyzer.go

```diff
@@ -241,6 +257,175 @@ func run(pass *analysis.Pass) (result interface{}, _ error) {
 	return Result{FullTriggers: triggers, Errors: errs}, nil
 }
 
+// duplicateFullTriggersFromContractedFunctionsToCallers duplicates all the full triggers that have
+// FuncParam producer or UseAsReturn consumer or both, from the contracted functions to the callers
+// of all the contracted functions. This is necessary because we have created new
+// producers/consumers for argument pass or result return at every call site of the contracted
+// function. In order to connect such producers/consumers back to the contracted functions, we
+// create new full triggers that duplicates the original full triggers of the contracted functions
+// but uses the new producers/consumers instead.
+func duplicateFullTriggersFromContractedFunctionsToCallers(
+	pass *analysis.Pass,
+	funcContracts functioncontracts.Map,
+	funcTriggers [][]annotation.FullTrigger,
+	funcResults map[*types.Func]*functionResult,
+) {
+
+	// Find all the calls to contracted functions
+	// callsByCtrtFunc is a mapping contracted function -> caller -> call expression
+	callsByCtrtFunc := map[*types.Func]map[*types.Func]*ast.CallExpr{}
+	for funcObj, r := range funcResults {
+		for ctrFunc, calls := range findCallsToContractedFunctions(r.funcDecl, pass, funcContracts) {
+			for _, call := range calls {
+				// TODO: have to introduce v to make NilAway happy. Remove v when NilAway can
+				//  tolerate this.
```

Comment:
Can we link an issue here explaining why `v` is needed? Or else add more detail to this comment on why this is needed now and why we think it might not be in the future?

Reply:
Enriched the code comment.

Ideally we should write this

```go
if _, ok := callsByCtrtFunc[ctrFunc]; !ok {
    callsByCtrtFunc[ctrFunc] = map[*types.Func]*ast.CallExpr{}
}
callsByCtrtFunc[ctrFunc][funcObj] = call // line 285
```

but NilAway would complain
```
Value read deeply from the variable `callsByCtrtFunc` lacking guarding at function/analyzer.go:285:5 (definitely nilable) and is written to at an index at function/analyzer.go:285:5 (must be nonnil)
```

Reply:
@sonalmahajan15 This _seems_ like it should be handled by trackable expressions. Is it some weird trackable expression + guarding interaction? Do we have a JIRA ticket on this we might want to migrate to an OSS issue :)

Reply:
Ah yes, this is a known problem, where tracking non-literals is limited. We do have a JIRA ticket for this, which we should now migrate to OSS. Will do :)